src/api/app/helpers/webui/notification_helper.rb

```diff
@@ -16,6 +18,75 @@ def filter_notification_link(link_text, amount, filter_item)
     end
   end
 
+  def notification_icon(notifiable_type)
+    return image_tag('request_icon', size: 18) if notifiable_type == 'BsRequest'
+
+    capture do
+      tag.i(class: 'fas fa-comments mr-1')
+    end
+  end
+
+  def request_state(notification)
+    return unless notification.notifiable_type == 'BsRequest'
+
+    state = notification.notifiable.state
+    color = request_badge_color(state)
+
+    capture do
+      tag.span(state, class: "badge badge-#{color}")
+    end
+  end
+
+  def request_badge_color(state)
+    case state
+    when :review, :new
+      'secondary'
+    when :declined, :revoke
+      'danger'
+    when :superseded
+      'warning'
+    when :accepted
+      'success'
+    else
+      'dark'
+    end
+  end
+
+  def notification_link(notification)
+    link_text = case notification.event_type
+                # TODO: add 'Event::ReviewWanted' when converted to BsRequest notification
+                when 'Event::RequestStatechange', 'Event::RequestCreate'
+                  "##{notification.notifiable.number}"
+                when 'Event::CommentForRequest'
+                  "Request ##{notification.notifiable.commentable.number}"
+                when 'Event::CommentForProject'
+                  notification.notifiable.commentable.name
+                when 'Event::CommentForPackage'
+                  commentable = notification.notifiable.commentable
+                  "#{commentable.project.name} / #{commentable.name}"
+                end
+
+    link_to(link_text, notification.link_to_notification_target, class: 'mx-1 text-word-break-all')
+  end
+
+  def avatars(notification)
```

Comment:
This *really* looks like a partial.

Reply:
True! I already moved as much as I could to a partial. Please have another look, @hennevogel